common/elasticsearch/src/main/scala/uk/ac/wellcome/elasticsearch/ElasticsearchIndexCreator.scala

```diff
@@ -43,6 +43,10 @@ class ElasticsearchIndexCreator(
           // If/when we index more records, we should revisit this.
           //
           .shards(1)
+          // Elasticsearch has a default maximum number of fields of 1000.
+          // Because images have all of the WorkData fields defined twice in the mapping,
+          // they end up having more than 1000 fields, so we increase them to 2000
+          .settings(Map("mapping.total_fields.limit" -> 2000))
```

Comment:
Another way around this is to store, but not index, non-searchable data.
https://www.elastic.co/guide/en/elasticsearch/reference/current/enabled.html

I've wanted to do it for the catalogue, as we do leak loads of model specific data into elastic.

Reply:
I agree that we should be keeping on top of how many fields we index. Did we work out why there was more than 1000 in the mapping?

Reply:
We talked about non-searchable data, but this would mean that if we want to make a field searchable we have to reindex. As the scale of our data isn't huge and ingestion times aren't an issue, we decided to index everything so we can iterate on how to query the data without needing reindexes. We can always make them non-searchable later if we find a reason. I don't think the 1000 limit is a good reason

As per the number of fields, I've logged the mapping, it's quite big. I didn't count them, but WorkData has way more than 100 fields

Reply:
Do you have a file with the mapping that I can see?

Reply:
If we're not actually interested in searching these fields, I think @jamesgorrie's suggestion of not indexing them is a good one.

Reply:
Yes, I uploaded to the slack channel in the thread, can't seem to figure out how to add it here..

Reply:
We're definitely interested in searching these fields - probably not all of them, but we don't know which ones we do want so we should have all of them searchable to make testing easier imo

Reply:
I think which fields are we going to search is a decision that will need iterating and testing. It would make our life easier if adding one field to the search query didn't mean reindexing. 100% we index stuff in workdata that has no use to search but that is probably cross purposes in works _and_images, so if we fix it once in  the works mapping, we should fix it for both and remove this line. But I think, for now, it's best if the code is reused even if that means a bit of waste at indexing time, rather than having a different mapping for the workdata on images to the mapping for workdata on works

Reply:
There is things in the mapping like `ontologyType` (appears 162 times) which I think are completely pointless to index as these are constants (I don't think there is any need for this to appear in the internal model at all, and should maybe just be in display, but that is another question).

I understand the arguments about indexing things we might at a future date need, but I reckon if there is anything like that we are sure we won't need I think we should not index them. Thoughts?

Reply:
Yes, definitely. Also, the "type" field is the circe type discriminator and I'm not even 100% sure when it's there and when it's not so definitely not going to be used for search, at least not is that many places. Might be useful on works but nowhere else. So 100% take out those that are never needed. 
I think it's something that should be done for both images _and_ works though so the approach in this PR to reuse the mapping code for WorkData mapping can be retained. So, I agree but I also think It's outside of the scope of this PR

Reply:
We use a `type` discriminator in one place for search apparently:
https://github.com/wellcomecollection/catalogue/blob/master/api/api/src/main/scala/uk/ac/wellcome/platform/api/services/WorksRequestBuilder.scala#L158-L163

I think I have found it useful filtering for e.g. `IdentifiedInvisibleWork` or what have you in Kibana, so there might be more of a use case there.

Agree beyond scope of this PR to remove `ontologyType` mapping.

Reply:
>  (I don't think there is any need for this to appear in the internal model at all, and should maybe just be in display, but that is another question)

I totally agree with that, I would really like to see it disappear from the internal model

Reply:
+1 to removing ontologyType!